File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int size = numbers.size();
        for(int x = 0; x < size; x++){
            int l = x;
            int r = size-1;
            while(numbers[l]+numbers[r] > target && r > l){
                r--;
            }
            while(numbers[l]+numbers[r] < target && r > l){
                l++;
            }
            if(numbers[l]+numbers[r] == target)
                return{l+1, r+1};
        }
    }
};

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-3.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        stack = []
        for num in arr[1:]:
            pops = 1
            while len(stack) and stack[-1] < num:
                pops += 1
                stack.pop()
            for pop in range(pops):
                stack.append(num)
        stack.append(-1)

        return stack
        

# Monotonic stack
"""

Check a number
if it is greater than the number on top
pop until it isn't x times
place x numbers on stack
place -1 to finish


"""